arrow-core/src/main/kotlin/arrow/core/extensions/set/semigroupal/SetKSemigroupal.kt

```diff
@@ -21,16 +21,16 @@ import kotlin.jvm.JvmName
 @Deprecated(
   "@extension kinded projected functions are deprecated",
   ReplaceWith(
-  "product(arg1)",
-  "arrow.core.product"
+    "product(arg1)",
+    "arrow.core.product"
```

Comment:
This operator is still missing atm. We should add `fun <A, B> Set<A>.zip(other: Set<B>): Set<Pair<A, B>>` and `fun <A, B, C> Set<A>.zip(other: Set<B>, f: (A, B) -> C): Set<C>` so we can deprecate this with `"this.zip(arg1, ::Tuple2)", "arrow.core.zip", "arrow.core.Tuple2"`

Reply:
Hey buddy you meant here adding that fun returning
```
fun <A, B> Set<A>.zip(other: Set<B>): Set<Pair<A, B>>
```
or
```
fun <A, B> Set<A>.zip(other: Set<B>): Set<Tuple2<A, B>>
```